File: dotnet/src/data_models/CorrelationResult.cs
namespace ProcessEngine.ConsumerAPI.Contracts
{
    /// <summary>
    /// Contains information about the result with which a correlation has finished execution.
    /// </summary>
    public class CorrelationResult<TPayload>
    {
        /// <summary>
        /// Contains the final result with which a correlation was finished.
        /// </summary>
        /// <value>The token payload.</value>
        public TPayload TokenPayload { get; set; }

        /// <summary>
        /// The Id of the correlation that was finished.
        /// </summary>
        public string CorrelationId { get; set; }

        /// <summary>
        /// The Id of the end event with which the correlation was finished.
        /// </summary>
        public string EndEventId { get; set; }
    }
}

File: typescript/src/data_models/correlation_result.ts
/**
 * Contains information about the result with which a correlation has finished execution.
 */
export class CorrelationResult {
  /**
   * The id of the correlation that was finished.
   */
  public correlationId: string;
  /**
   * The id of the EndEvent with which the correlation was finished.
   */
  public endEventId: string;
  /**
   * Contains the final result with which a correlation was finished.
   */
  public tokenPayload: string;
}
